File: 2st/Riyeon/명예의 전당 (1).java
import java.util.*;
class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        for(int i=0; i<score.length; i++){
            int[] arr = Arrays.copyOfRange(score, 0, i+1);
            Arrays.sort(arr);
            if(arr.length < k)
                answer[i] = arr[0];
            else
                answer[i] = arr[arr.length-k];
        }
        return answer;
    }
}

File: 2st/Riyeon/명예의 전당 (1).py
def solution(k, score):
    answer = []
    li = []
    for i in score:
        li.sort()
        if len(li) >= k:
            if li[0] < i:
                li.pop(0)
                li.append(i)
        else:
            li.append(i)
        answer.append(min(li))
    return answer